src/mesh/generic/Boundary.cpp

```diff
@@ -109,19 +114,17 @@ void Boundary::generate_sinusoidal(json const& boundary,
 
         auto const scale_factor = number_of_cycles[i] * period[i] / time_block.back();
 
-        time_block = time_block | view::transform([scale_factor, new_times](double x) {
-                         return x * scale_factor + new_times.back();
+        time_block = time_block | view::transform([scale_factor, times](double x) {
```

Comment:
Only the last entry should be captured by the lambda otherwise we're taking a copy of the times array when we only want the .back() element right?

Reply:
True. Capturing by reference solves the problem, right?